scipy/optimize/tests/test_optimize.py

```diff
@@ -226,6 +226,41 @@ def test_powell(self):
                          [1.72949016, -0.44156936, 0.47576729]],
                         atol=1e-14, rtol=1e-7)
 
+    def test_powell_bounded(self):
+        # Powell (direction set) optimization routine
+        # same as test_powell above, but with bounds
```

Comment:
If I'm reading correctly, it looks like the bounds do not affect the optimal solution. Is that right?
If so, there should be at least one test that shows that adding bounds changes the optimal solution.

Reply:
Yes this is the same test being used for the `test_powell` function right above. I will implement a test soon where bounded Powell returns a different solution than Powell. I will recomment here once that is complete.

Reply:
I implemented a few more tests. 

One test is where the global minimum is outside the bounds, Powell finds the global minimum but the bounded Powell method finds the minimum that is within the bounds. 

One test is where the global minimum is within the bounds, but Powell only finds a local minimum whereas the bounded Powell method finds the global minimum.

One test is where the initial guess is outside the bounds. I check to make sure that an ``OptimizeWarning` is warned, and that the final result is within the bounds.

One test is where the initial guess is outside the bounds and the `direc` argument is modified such that the parameter that is outside the bounds is never modified. In this case, the final remains outside the bounds. I check to make sure that the `status` and `success` are set appropriately.